document-portal/document-portal-fuse.c

```diff
@@ -1428,7 +1428,7 @@ xdp_fuse_getattr (fuse_req_t req,
   double attr_valid_time = 0.0;/* Time in secs for attribute validation */
   const char *op = "GETATTR";
 
-  g_debug ("GETATTR %lx", ino);
+  g_debug ("GETATTR %" G_GINT64_MODIFIER "x", ino);
```

Comment:
Is `fuse_ino_t` guaranteed to be signed 64-bit on all architectures?

(If not, then this would be wrong, but `"%llx", (long long) ino` would work.)

Reply:
Hi Simon, long time no see :-)

No, it's actually **unsigned** in all architectures: https://github.com/libfuse/libfuse/blob/master/include/fuse_lowlevel.h#L46

But there isn't a `G_GUINT64_MODIFIER` macro, and if my understanding is correct the reason is that this macro only contains the length modifier part for the sprintf operation, not the actual type, so it can be used in both cases (for 32 bit types, it expands to `ll`). I'm not sure why you think that `ino` requires a cast, since the `x` conversion specifier expects an unsigned integer, as `ino` is.

Reply:
Ah, maybe your question was about the bit length rather than about the signedness. :-)

Reply:
Ah, yes, I should have said: is it guaranteed to be unsigned 64-bit on all architectures? (And the important part is the width, not the signedness.)

Reply:
Please mention in the commit message that this is a FUSE 3 change, rather than just saying "use portable specifiers": `%lx` was the correct, portable specifier for FUSE 2, but isn't correct any more for FUSE 3.

Reply:
Thanks, commits updated.